reactor-core/src/test/java/reactor/core/publisher/MonoElementAtTest.java

```diff
@@ -247,4 +247,26 @@ public void scanSubscriber() {
 		test.cancel();
 		assertThat(test.scan(Scannable.Attr.CANCELLED)).isTrue();
 	}
+
+	@Test
```

Comment:
cool!

for these two tests, let's use more up to date best practices: can you try to use `StepVerifier.create(...)` instead of `AssertSubscriber`?

The `StepVerifier` can also be used with `AssertJ` to both assert the exception type AND the message, which I think would make it interesting to add a case for the `empty` source as well.

(also, you can remove the `public` modifier in new tests... we're now using JUnit5, despite the older legacy test still having the public modifier, it is redundant and should at lest be avoided in newly written tests).

Reply:
updated the test cases, please take a look.
thank you!